python/py-avro/Portfile

```diff
@@ -4,10 +4,11 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           python 1.0
 
-github.setup        apache avro 1.9.1 release-
+github.setup        apache avro 1.10.0 release-
 revision            0
 name                py-${github.project}
 categories-append   devel
+fetch.type          git
```

Comment:
why did you add this - it seem to work perfectly fine without it

Reply:
Now for me. Here's what I see:

> `sudo port extract py37-avro`
```
:info:extract Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.9.1.tar.gz' | /usr/bin/tar -xf -
:debug:extract system:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.9.1.tar.gz' | /usr/bin/tar -xf -
:info:extract apache-avro-8921826/lang/java/mapred/src/test/resources/org/apache/avro/mapreduce/mapreduce-test-input.avro/.part-r-00000.avro.crc: Failed to restore metadata
:info:extract tar: Error exit delayed from previous errors.
```

Reply:
this suggests that you're using the old distfile (the name above say `avro-1.9.1.tar.gz`), clean the port first and try again after removing the `fetch.type` stuff and revert to the using `worksrcdir          ${distname}/lang/py3`

Reply:
Nope:

> `sudo port clean --all py37-avro ; sudo port extract py37-avro`
```
:info:extract Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.9.1.tar.gz' | /usr/bin/tar -xf -
:debug:extract system:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.9.1.tar.gz' | /usr/bin/tar -xf -
:info:extract apache-avro-8921826/lang/java/mapred/src/test/resources/org/apache/avro/mapreduce/mapreduce-test-input.avro/.part-r-00000.avro.crc: Failed to restore metadata
:info:extract tar: Error exit delayed from previous errors.
:info:extract Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.9.1.tar.gz' | /usr/bin/tar -xf -
:info:extract Exit code: 1
:error:extract Failed to extract py37-avro: command execution failed
```

Reply:
This fail is with the current Portfile, https://github.com/macports/macports-ports/blob/master/python/py-avro/Portfile

Reply:
we don't really care what happens with the old Portfile, this PR is about how to update to the latest version. For that you can just update the `version`, `checksums`, and dependencies while keeping the `worksrcdir`.

Reply:
> just update the `version`, `checksums`, and dependencies while keeping the `worksrcdir`.

@reneeotten That **will not work*** because of the extract issue described above. This is a bugfix. The only extract that I'm aware of that works in this repo is `fetch.type git`.

Reply:
I don't know if you actually tried any of my suggestions, but the Portfile below works just fine:

```
> port -dv installed py38-avro
The following ports are currently installed:
  py38-avro @1.10.0_0 (active) platform='darwin 18' archs='x86_64' date='2020-07-08T16:03:53-0400'
```

```
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

PortSystem          1.0
PortGroup           github 1.0
PortGroup           python 1.0

github.setup        apache avro 1.10.0 release-
revision            0
name                py-${github.project}
categories-append   devel

platforms           darwin
license             Apache-2
maintainers         nomaintainer

description         Avro is a serialization and RPC framework.
long_description    ${description}

homepage            https://avro.apache.org/

checksums           rmd160  e398cc448a227de8754eb60848eb4a47f0633265 \
                    sha256  4e94b43ebdb1081fd842644bda2caaddc586c1b608efe9236d348d01bf0866a4 \
                    size    1899682

python.versions     37 38

if {${name} ne ${subport}} {
    depends_build-append \
                    port:py${python.version}-setuptools \
                    port:py${python.version}-codestyle \
                    port:py${python.version}-isort

    worksrcdir      ${worksrcpath}/lang/py3

    test.run        yes
    test.cmd        py.test-${python.branch}
    test.target
    test.env-append PYTHONPATH=${worksrcpath}/build/lib

    post-destroot {
        set docdir ${prefix}/share/doc/${subport}
        xinstall -d ${destroot}${docdir}
        xinstall -m 0644 -W ${worksrcpath}/../../ LICENSE.txt README.md \
            ${destroot}${docdir}
    }

    livecheck.type      none
} else {
    github.livecheck.regex {([0-9.]+)}
}
```

Reply:
This does not work for me, same as all the comments above. There's an issue with the repo. Using `fetch.type git` fixes it.

> `sudo port clean --all py37-avro ; sudo port extract py37-avro`
```
:info:extract Executing:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.10.0.tar.gz' | /usr/bin/tar -xf -
:debug:extract system:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.10.0.tar.gz' | /usr/bin/tar -xf -
:info:extract apache-avro-f83cd83/lang/java/mapred/src/test/resources/org/apache/avro/mapreduce/mapreduce-test-input.avro/.part-r-00000.avro.crc: Failed to restore metadata
:info:extract tar: Error exit delayed from previous errors.
:info:extract Command failed:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-avro/py37-avro/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/py-avro/avro-1.10.0.tar.gz' | /usr/bin/tar -xf -
:info:extract Exit code: 1
```

Reply:
macOS 10.15.5 19F101
Xcode 11.5 11E608c

Reply:
Moved to pypi.